### PR TITLE
chore: rebase develop release v2.0.0

### PR DESCRIPTION
**Pull request by kpark277**
Description of changes:
* Rebase release/v2.0.0 into develop

Conflicts that had to be manually resolved:
* both modified:   .github/workflows/merge-develop-to-stage.yml
* both modified:   .viperlightignore
* both modified:   common/config/rush/pnpm-lock.yaml
* both modified:   common/config/rush/repo-state.json
* both modified:   solutions/swb-app/config/jest.config.json
* both modified:   solutions/swb-app/src/generateRouter.ts
* both modified:   solutions/swb-reference/src/RSWStack.ts
* both modified:   workbench-core/authentication/src/authenticationMiddleware.ts
* both modified:   workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.test.ts
* both modified:   workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts
* both modified:   workbench-core/example/infrastructure/integration-tests/support/complex/datasetHelper.ts
* both modified:   workbench-core/user-management/src/plugins/cognitoUserManagementPlugin.test.ts

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by aws-tyler — APPROVED** [review]

File: solutions/swb-app/src/generateRouter.ts [hunk 1]
```diff
@@ -50,14 +51,21 @@ export function generateRouter(apiRouteConfig: ApiRouteConfig): Express {
...
   const cognitoPluginOptions: CognitoAuthenticationPluginOptions = {
     cognitoDomain: process.env.COGNITO_DOMAIN!,
     userPoolId: process.env.USER_POOL_ID!,
     webUiClient: {
       clientId: process.env.WEB_UI_CLIENT_ID!,
       clientSecret: process.env.WEB_UI_CLIENT_SECRET!
     },
-    allowedClientIds: [process.env.PROGRAMMATIC_ACCESS_CLIENT_ID!]
+    allowedClientIds: [process.env.PROGRAMMATIC_ACCESS_CLIENT_ID!],
+    tokenRevocationService
```
Comment (janeyuaws) [inline solutions/swb-app/src/generateRouter.ts:68]: can you double check if the conflict is resolved in the expected way. it doesn't match current release branch

File: workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts [hunk 2]
```diff
@@ -89,12 +97,13 @@ export class CognitoAuthenticationPlugin implements AuthenticationPlugin {
     cognitoDomain,
     userPoolId,
     webUiClient,
-    allowedClientIds = []
+    allowedClientIds = [],
+    tokenRevocationService
```
Comment (janeyuaws) [inline workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts:101]: same here

**Review by janeyuaws — APPROVED** [review]

File: solutions/swb-app/src/generateRouter.ts (hunk 1 above)
Comment (kpark277) [reply solutions/swb-app/src/generateRouter.ts:68]: Confirmed that this is from updates made on the develop branch.

File: workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts (hunk 2 above)
Comment (kpark277) [reply workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts:101]: Confirmed that the list of client Ids is related to a new feature that was added to develop.

File: workbench-core/example/infrastructure/integration-tests/support/complex/datasetHelper.ts [hunk 3]
```diff
@@ -54,14 +58,26 @@ export class DatasetHelper {
   }
 
   public static async deleteDdbRecords(awsService: AwsService, dataSetId: string): Promise<void> {
-    await awsService.helpers.ddb
-      .delete({ pk: `${dataSetPrefix}#${dataSetId}`, sk: `${dataSetPrefix}#${dataSetId}` })
-      .execute();
+    // delete dataset entry
```
[the remark this answers is not in the document]
Comment (kpark277) [reply workbench-core/example/infrastructure/integration-tests/support/complex/datasetHelper.ts:61]: Confirmed that this is from updates made on the develop branch.

**Review by insignias — APPROVED** [review]

**Review by htztomic — APPROVED** [review]

**Pull request merged**